sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -798,3 +901,11 @@ case class GreaterThanOrEqual(left: Expression, right: Expression)
 
   protected override def nullSafeEval(input1: Any, input2: Any): Any = ordering.gteq(input1, input2)
 }
+
+/**
+ * Only be used in PredicateSubquery, and will be rewrite as Not(EqualTo(left, right)).
+ */
+case class NotEqualTo(left: Expression, right: Expression) extends BinaryComparison {
```

Comment:
We don't like this. Could you look for another approach?

Reply:
OK, I'll try to parse `x != ANY(query)` to `Not(AnySubquery(x, EqualTo, query))`, so that we won't need it any more.